tests/console/hwloc_testsuite.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's hwloc tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run hwloc testsuite
+# Maintainer: Thomas Blume <tblume@suse.com>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+
+sub run() {
+    # install hwloc testsuite
+    script_run('zypper ar http://download.suse.de/ibs/QA:/SLE12SP2/update/ hwloc-testrepo');
+    script_run('zypper ar http://download.suse.de/ibs/Devel:/HPC:/SLE12SP2/standard/ HPC-module-hwloc');
+    script_run('zypper --gpg-auto-import-keys ref');
+    assert_script_run('zypper -n in hwloc hwloc-testsuite');
+
+    # run the testsuite test scripts
+    assert_script_run('cd /var/opt/hwloc-tests && ./run-tests.sh');
+
+    # capture a screenshot and compare with needle 'hwloc_testsuite-hwloc-testsuite-result-20170420'
```

Comment:
I guess you can delete that comment. I know, it's new for you but I will become second nature that `assert_screen` means exactly that in no time ;-)